chat-core/src/assertEnvVars.ts

```diff
@@ -7,6 +10,11 @@ export const assertEnvVars = <T extends Record<string, string>>(
       // Get this var value from process.env - either valid string or undefined
       vars[currentVar] = process.env[currentVar];
 
+      // if the var is a url, make sure it has a following slash
+      if (URL_ENV_VARS.includes(currentVar)) {
```

Comment:
assertEnvVars is agnostic to exact env var names (it is used it other parts). We would need another way of determining what the URLs are supposed to be.

Can we instead assert for trailing slashes where the vars are actually used?